src/main/scala/com/workday/warp/junit/HasTestId.scala

```diff
@@ -0,0 +1,34 @@
+package com.workday.warp.junit
+
+import java.lang.reflect.Method
+
+/** Logic for constructing a testId given a testClass and testMethod.
+  *
+  * Multiple Junit interfaces [[org.junit.jupiter.api.extension.ExtensionContext]] and [[org.junit.jupiter.api.TestInfo]],
+  * for example, declare methods `getTestClass` and `getTestMethod`, but share no common supertype.
+  *
+  * We use ad-hoc polymorphism to declare [[HasTestId]] instances for [[org.junit.jupiter.api.TestInfo]] and
+  * [[org.junit.jupiter.api.extension.ExtensionContext]].
+  *
+  * Created by tomas.mccandless on 6/18/20.
+  */
+trait HasTestId {
+
+  def getTestClass: Option[Class[_]]
+
+  def getTestMethod: Option[Method]
+
+  /**
+    * Attempts to construct a fully qualified method name.
+    *
+    * TODO make sure this works for dynamic tests
```

Comment:
Is this TODO still valid? Does this need more testing before it's ready to be merged?

Reply:
yep this is still valid. i think this pr is mergeable but not necessarily releasable.

dynamic tests don't participate in normal junit test lifecycle (BeforeAll, BeforeEach, etc) so measuring them will be slightly trickier. i will double check whether dynamic tests even have a class and method, that was really my only concern with this TODO comment.